### ISSUE TITLE
[FE] My VA: Investigate payment info feature toggle behavior

### ISSUE THREAD

**Issue by ajakabcin** [department-of-veterans-affairs/va.gov-team#58543]
## Background
While looking into #57991 , we found that users may somehow still be seeing and clicking on links from payment info mvp - where a single section for "Benefit Payments and Debts" exists vs payment info v2 which should be live for all users and has two distinct sections for "Outstanding Debts" and "Benefit Payments". 

We need to investigate how and why this is happening since the feature is set as enabled to 100% of users in production.

## Tasks
- [x] Investigate why users are somehow able to access the old version of payment info (all users should be seeing the new version, and only the new version)
- [x] Identify a solution to resolve this

## Acceptance Criteria
- We have a solution to resolve the issue

**Comment by allisonlu** [comment]
@ajakabcin 

## problem
I _think_ the following are the culprits:

- how `showBenefitPaymentsAndDebt` is defined too broadly. It's essentially only checking if user is LOA3, and there is no error connecting to the MPI
  ```
  const showBenefitPaymentsAndDebt =
    !showMPIConnectionError && !showNotInMPIError && isLOA3;
  ```
  - source: [lines 347–348](https://github.com/department-of-veterans-affairs/vets-website/blob/363c44eb1b1517c4ea0136722c8a89ed95cd9e61/src/applications/personalization/dashboard/components/Dashboard.jsx#LL347C9-L347C35)

- because of that, there are many opportunities for the following conditional to be true and display the old Benefit payments and debts section
  ```
  {canAccessPaymentHistory &&
   showBenefitPaymentsAndDebt &&
   !showBenefitPaymentsAndDebtV2 ? (
       <BenefitPaymentsAndDebt
          ...
        />
  ) : null}
  ```
  - source: [lines 259–266](https://github.com/department-of-veterans-affairs/vets-website/blob/363c44eb1b1517c4ea0136722c8a89ed95cd9e61/src/applications/personalization/dashboard/components/Dashboard.jsx#LL259C1-L266C24)

- the conditional to display the separate Benefit payments and Outstanding debts sections **only** checks if `showBenefitPaymentsAndDebt` (shown above) is true and the feature toggle is turned on:
  ```
  {showBenefitPaymentsAndDebtV2 ? (
     <>
       <DebtsV2 />
       <BenefitPaymentsV2
         ...
       />
     </>
   ) : null}
  ```
  - source: [lines 267–275](https://github.com/department-of-veterans-affairs/vets-website/blob/363c44eb1b1517c4ea0136722c8a89ed95cd9e61/src/applications/personalization/dashboard/components/Dashboard.jsx#LL267C1-L275C24)

## possible solutions

1. We can combine the conditionals into one:
```
{canAccessPaymentHistory &&
 showBenefitPaymentsAndDebt &&
!showBenefitPaymentsAndDebtV2 ? (
  <BenefitPaymentsAndDebt
  ...
  />
 ) : (
   <>
      <DebtsV2 />
      <BenefitPaymentsV2
      ...
      />
   </>
)}
```

2. or completely remove it (since no one should be seeing the old Benefit payments and debts section anyway right?)

**Comment by ajakabcin** [comment]
Closing - solution 2 that Allison outlined above will be implemented in #47243
